Challenge-4/Bot_2/src/helper.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  CONFIGURATION_ABI,
+  CONFIGURATOR_PROXY,
+  CONFIGURATOR_PROXY_POLYGON,
+} from "./constants";
+import { AlertsResponse, ethers, getAlerts } from "forta-agent";
+import { AlertType } from "./types";
+import { LRUCache } from "lru-cache";
+
+export default class Helper {
+  private provider: ethers.providers.Provider;
+  constructor(provider: ethers.providers.Provider) {
+    this.provider = provider;
+  }
+
+  public secondsPerYear = 31536000;
+
+  public async getProtocolConfiguration(
+    tokenAddress: string ,
+    configuratorProxy: string,
+    blockNumber: number,
+  ): Promise<any> {
+      const configuration = new ethers.Contract(
+      configuratorProxy,
+      CONFIGURATION_ABI,
+      this.provider,
+    );
+
+    return await configuration.callStatic.getConfiguration(tokenAddress, {
+      blockTag: blockNumber,
+    });
+  }
+
+  public async getUtilization(
+    tokenAddress: string,
+    abi: any,
+    blockNumber: number,
+  ) {
+    const utilizationContract = new ethers.Contract(
+      tokenAddress,
+      [abi],
+      this.provider,
+    );
+    return await utilizationContract.callStatic.getUtilization({
+      blockTag: blockNumber,
+    });
+  }
+
+  // getSupplyRate returns the supply rate in terms of seconds. It has to be converted to annual rate
+  public async getSupplyAPR(
```

Comment:
Same: caching

Reply:
supplyAPR might change more frequently as the supply rate rate is calculated per second. Should I still go with caching?